src/Features/Navigation/Menu.php

```diff
@@ -506,21 +511,44 @@ public function migrate_core_child_items( $menu ) {
 				continue;
 			}
 
+			$menu_item[2] = htmlspecialchars_decode( $menu_item[2] );
+
 			// Don't add already added items.
 			$callbacks = self::get_callbacks();
 			if ( array_key_exists( $menu_item[2], $callbacks ) ) {
 				continue;
 			}
 
-			self::add_item(
+			// Don't add these Product submenus because they are added elsewhere.
+			if ( in_array( $menu_item[2], array( 'product_importer', 'product_exporter', 'product_attributes' ), true ) ) {
+				continue;
+			}
+
+			self::add_plugin_item(
 				array(
-					'parent'     => 'woocommerce-settings',
 					'title'      => $menu_item[0],
 					'capability' => $menu_item[1],
 					'id'         => sanitize_title( $menu_item[0] ),
 					'url'        => $menu_item[2],
 				)
 			);
+
+			// Determine if migrated items are a taxonomy or post_type. If they are, register them.
+			$parsed_url   = wp_parse_url( $menu_item[2] );
+			$query_string = isset( $parsed_url['query'] ) ? $parsed_url['query'] : false;
+
+			if ( $query_string ) {
+				$query = array();
+				parse_str( $query_string, $query );
+
+				if ( isset( $query['taxonomy'] ) ) {
+					Screen::register_taxonomy( $query['taxonomy'] );
```

Comment:
The only way I can think of to break this is if a taxonomy was part of WooCommerce, but the associated post type was not.  E.g. a `product_label` taxonomy was added to the default `post` post type which would register `post` as a WooCommerce screen since the URL for taxonomies includes the post type `wp-admin/term.php?taxonomy=product_label&tag_ID=209&post_type=post`

This seems pretty edge case, but we could fix by doing an if/else on this logic.  I think we're also fairly safe as is, so I'll leave this up to you.